services/app-web/src/components/FilterAccordion/FilterAccordion.test.tsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react'
+
+describe('FilterAccordion', () => {
+    const stateOptions = [
+        { label: 'Alaska', value: 'AK' },
+        { label: 'New York', value: 'NY' },
+        { label: 'California', value: 'CA' },
+        { label: 'Ohio', value: 'OH' },
+    ]
+
+    const cityOptions = [
+        { label: 'Juneau', value: 'Juneau' },
+        { label: 'Albany', value: 'Albany' },
+        { label: 'Sacramento', value: 'Sacramento' },
+        { label: 'Cleveland', value: 'Cleveland' },
+    ]
+
+    const countryOptions = [
+        { label: 'United States', value: 'United States' },
+        { label: 'Canada', value: 'Canada' },
+        { label: 'Mexico', value: 'Mexico' },
+        { label: 'Brazil', value: 'Brazil' },
+    ]
+
+    const filters = [
+        <FilterSelect
+            filterOptions={stateOptions}
+            name="state"
+            label="State"
+            key="state"
+        />,
+        <FilterSelect
+            filterOptions={cityOptions}
+            name="city"
+            label="City"
+            key="city"
+        />,
+        <FilterSelect
+            filterOptions={countryOptions}
+            name="country"
+            label="Country"
+            key="country"
+        />,
+    ]
+
+    it('renders filter accordion and expected filters', async () => {
+        renderWithProviders(
+            <FilterAccordion
+                filterTitle={'Testing filter accordion'}
+                children={filters}
+            />,
+            {
+                apolloProvider: {
+                    mocks: [
+                        fetchCurrentUserMock({
+                            statusCode: 200,
+                        }),
+                    ],
+                },
+            }
+        )
+
+        expect(screen.getByTestId('accordion')).toBeInTheDocument()
+        expect(screen.getByText('Testing filter accordion')).toBeInTheDocument()
+        expect(
+            screen.getByTestId('accordionButton_filterAccordionItems')
+        ).toBeInTheDocument()
+
+        const accordionButton = screen.getByTestId(
+            'accordionButton_filterAccordionItems'
+        )
+        await waitFor(async () => {
+            expect(screen.queryByTestId('accordion')).toBeInTheDocument()
+            await userEvent.click(accordionButton)
+        })
+
+        expect(screen.getByTestId('state-filter')).toBeInTheDocument()
+        expect(screen.getByTestId('city-filter')).toBeInTheDocument()
+        expect(screen.getByTestId('country-filter')).toBeInTheDocument()
+        expect(
+            screen.getByRole('button', { name: 'Clear filters' })
+        ).toBeInTheDocument()
+    })
+    it('clears all filters when Clear filters button is clicked', async () => {
+        renderWithProviders(
+            <FilterAccordion
+                filterTitle={'Testing filter accordion'}
+                children={filters}
+            />,
```

Comment:
Here I'm passing an array of filter components in the `children` prop, instead of nesting inside the component. The `renderWithProviders` does not like it when I nest the `FilterSelect` inside the `FilterAccordion`. It has to do with the way I'm using `React.cloneElement` in `FilterAccordion` to pass in `toggleClearFilter` prop to its children `FilterSelect`.